### PR TITLE
Refactor Seamless File Handling

### PR DESCRIPTION
**Pull request by hpohekar**
- [x] Remove upload and download methods from `BaseSession`.
- [x] Keep constructor argument `remote_file_handler: Optional[Any] = None`
- [x] Test Meshing session in the Ansys Lab
- [x] Test Solver session in the Ansys Lab
- [x] Implement design change related to `launcher.py`.
- [x] Test local Meshing session
- [x] Test local Solver session

### CODE REVIEW

**Comment by hpohekar** [comment]
@seanpearsonuk This PR is ready now.

File: src/ansys/fluent/core/session_meshing.py [hunk 1]
```diff
@@ -19,19 +19,27 @@ class Meshing(PureMeshing):
...
 
         Args:
             fluent_connection (:ref:`ref_fluent_connection`): Encapsulates a Fluent connection.
+            remote_file_handler: Supports file upload and download.
         """
-        super(Meshing, self).__init__(fluent_connection=fluent_connection)
+        super(Meshing, self).__init__(
+            fluent_connection=fluent_connection, remote_file_handler=remote_file_handler
+        )
+        self._remote_file_handler = remote_file_handler
```
Comment (mkundu1) [inline src/ansys/fluent/core/session_meshing.py:33]: This should already be set from the base session

File: src/ansys/fluent/core/session.py [hunk 2]
```diff
@@ -107,8 +107,8 @@ def __init__(
             fluent_connection (:ref:`ref_fluent_connection`): Encapsulates a Fluent connection.
             remote_file_handler: Supports file upload and download.
         """
-        self._remote_file_handler = remote_file_handler
         BaseSession.build_from_fluent_connection(self, fluent_connection)
+        self._remote_file_handler = remote_file_handler
```
Comment (mkundu1) [inline src/ansys/fluent/core/session.py:111]: Can we set this inside `build_from_fluent_connection`?

**Comment by mkundu1** [comment]
Overall looks good

File: src/ansys/fluent/core/session_meshing.py (hunk 1 above)
Comment (hpohekar) [reply src/ansys/fluent/core/session_meshing.py:33]: Right,  We will remove this. Thank you.

File: src/ansys/fluent/core/session.py (hunk 2 above)
Comment (hpohekar) [reply src/ansys/fluent/core/session.py:111]: Actually, prior to current change it was here but then we moved it to `__init__`. Yes we can set it here.

File: src/ansys/fluent/core/session_meshing.py (hunk 1 above)
Comment (hpohekar) [reply src/ansys/fluent/core/session_meshing.py]: Done. Thank you.

File: src/ansys/fluent/core/session.py (hunk 2 above)
Comment (hpohekar) [reply src/ansys/fluent/core/session.py]: Done. Thank you.

**Review by mkundu1 — APPROVED** [review]

**Review by raph-luc — APPROVED** [review]

File: src/ansys/fluent/core/launcher/launcher.py [hunk 3]
```diff
@@ -316,16 +320,25 @@ def launch_remote_fluent(
...
+
+    remote_file_handler = RemoteFileHandler(
+        transfer_service=PimFileTransferService(
+            pim_instance=fluent_connection._remote_instance
         )
     )
 
+    return session_cls(
+        fluent_connection=fluent_connection, remote_file_handler=remote_file_handler
+    )
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/launcher/launcher.py:340]: 💯

**Pull request merged**